compiler/src/dotty/tools/dotc/transform/CheckJVMSpecification.scala

```diff
@@ -0,0 +1,43 @@
+package dotty.tools.dotc
+package transform
+
+import MegaPhase.MiniPhase
+import GenericSignatures.*
+import core.Contexts.*
+import core.Decorators.*
+import core.Constants.*
+import core.Symbols.*
+import util.SrcPos
+
+final class CheckJVMSpecification extends MiniPhase:
+  import ast.tpd.*
+
+  override def phaseName: String = CheckJVMSpecification.name
+  override def description: String = CheckJVMSpecification.description
+
+
+  override def transformDefDef(tree: DefDef)(using Context): Tree =
+    val sym = tree.symbol
+    val memberTpe = atPhase(ctx.base.erasurePhase) { sym.owner.denot.thisType.memberInfo(sym) }
+    for sig <- GenericSignatures.javaSig(sym, memberTpe) do
```

Comment:
Does this mean we'll compute the Java generic signature twice for every `DefDef`? Once here and once in the actual back-end? That's not great; it's a somewhat expensive operation.